src/providers/VoiceFocusProvider/docs/useVoiceFocus.stories.mdx

```diff
@@ -72,17 +74,22 @@ const MyChild = () => {
   }, [device]);
 
   useEffect(() => {
-    let current = device;
-    if (isVoiceFocusOn) {
-      const currentDevice = audioInputSelectionToDevice(device);
-      current = addVoiceFocus(currentDevice);
-    } else {
-      if (device instanceof VoiceFocusTransformDevice) {
-        current = device.getInnerDevice();
+    async function onVFCheckboxChange() {
+      let current = device;
+      if (isVoiceFocusOn) {
+        if (typeof (device) === 'string') {
+          current = await addVoiceFocus(device);
+        }
+      } else {
+        if (device instanceof VoiceFocusTransformDevice) {
+          current = device.getInnerDevice();
+        }
       }
+      await meetingManager.selectAudioInputDevice(current);
     }
-    meetingManager.selectAudioInputDevice(current);
-  }, [isVoiceFocusOn];
+    
+    onVFCheckboxChange();
```

Comment:
The example does not mention in any checkbox UI element so would it be better if we change this method name?

Reply:
Yep, I can rename it